stylegan_code_finder/networks/trans_u_net/utils.py

```diff
@@ -27,16 +44,66 @@ def _dice_loss(self, score, target):
         loss = 1 - loss
         return loss
 
-    def forward(self, inputs, target, weight=None, softmax=False):
+
+class TverskyLoss(LossFunction):
+    def __init__(self, n_classes: int, alpha: float, beta: float):
+        super().__init__(n_classes)
+        self.alpha = alpha
+        self.beta = beta
+
+    def _metric_calculation(self, score: torch.Tensor, target: torch.Tensor) -> float:
+        target = target.float()
+        smooth = 1e-5
+        tp = torch.sum(score * target)
+        fp = torch.sum((1-target) * score)
+        fn = torch.sum(target * (1-score))
+        loss = (tp + smooth) / (tp + self.alpha*fp + self.beta*fn + smooth)
+        loss = 1 - loss
+        return loss
+
+
+class HandwritingLoss(LossFunction):
```

Comment:
Not sure if this class is really needed. In the end, it's just a special case of `LossFunction`. Couldn't the original class just be reused with `self.n_classes = 1` and the input tensor reduced to only the handwriting class.  
This solution would also remove the multiple inheritance further down.